src/main/java/practice/StreamPractice.java

```diff
@@ -75,6 +108,10 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates.stream()
+                .filter(c -> new CandidateValidator().test(c))
```

Comment:
```suggestion
                .filter(new CandidateValidator())
```
you can simplify it, but it better makes the CandidateValidator class-level variable